pkg/network/usm/postgres_monitor_test.go

```diff
@@ -889,3 +893,142 @@ func createFragment(fragment []byte) [ebpf.BufferSize]byte {
+		t.Run(testName, func(t *testing.T) {
+
+			expectedBuckets[i] = true
+			cleanProtocolMaps(t, "postgres", monitor.ebpfProgram.Manager.Manager)
+
+			require.NoError(t, monitor.Resume())
+			runQueryForBucket(t, pgClient, i*ebpf.MsgCountBucketSize)
+			require.NoError(t, monitor.Pause())
+
+			validateKernel(t, monitor, isTLS, expectedBuckets)
+			expectedBuckets[i] = false
+		})
+	}
+}
+
+func setupPGClient(t *testing.T, serverAddress string, isTLS bool) *postgres.PGXClient {
+	pg, err := postgres.NewPGXClient(postgres.ConnectionOptions{
+		ServerAddress: serverAddress,
+		EnableTLS:     isTLS,
+	})
+	require.NoError(t, err)
+	require.NoError(t, pg.Ping())
+	return pg
+}
+
+// createLargeTable It runs a postgres query to create a table large enough to retrieve long responses later.
+func createLargeTable(t *testing.T, pg *postgres.PGXClient, tableValuesCount int) {
+	require.NoError(t, pg.RunQuery(createTableQuery))
+	require.NoError(t, pg.RunQuery(createInsertQuery(generateTestValues(0, tableValuesCount)...)))
+}
+
+// runQueryForBucket The client sends a SELECT query to postgres with the specified SQL row limit of <N>.
+// For each SELECT request of <N> rows, the TCP response packet from Postgres is expected to contain
+//
+//	4 + <N> postgres messages in the format:
+//
+// "Bind completion", "Row description", "data row" 1...N, "command completion", "ready for query".
+func runQueryForBucket(t *testing.T, pg *postgres.PGXClient, limitCount int) {
+	require.NoError(t, pg.RunQuery(generateSelectLimitQuery(limitCount)))
+}
+
+// validateKernel Checking telemetry data received for a postgres query
+func validateKernel(t *testing.T, monitor *Monitor, tls bool, expected [ebpf.MsgCountNumBuckets]bool) {
+	var actual *ebpf.PostgresKernelMsgCount
+	assert.Eventually(t, func() bool {
+		found, err := getKernelTelemetry(monitor, tls)
+		require.NoError(t, err)
+		require.NotNil(t, found)
+		actual = found
+		return compareMessagesCount(found, expected)
+	}, time.Second*2, time.Millisecond*100)
+	if t.Failed() {
+		t.Logf("\nexpected telemetry:\n %+v;\nactual telemetry:\n %+v", expected, actual)
+	}
+}
+
+// getKernelTelemetry returns statistics obtained from the kernel
+func getKernelTelemetry(monitor *Monitor, isTLS bool) (*ebpf.PostgresKernelMsgCount, error) {
+	pgKernelTelemetry := &ebpf.PostgresKernelMsgCount{}
+	var zero uint32
+
+	mapName := postgres.KernelTelemetryMapPlain
+	if isTLS {
+		mapName = postgres.KernelTelemetryMapTLS
+	}
+	mp, _, err := monitor.ebpfProgram.GetMap(mapName)
+	if err != nil {
+		return nil, fmt.Errorf("unable to get %q map: %s", mapName, err)
+	}
+	if err := mp.Lookup(unsafe.Pointer(&zero), unsafe.Pointer(pgKernelTelemetry)); err != nil {
+		return nil, fmt.Errorf("unable to lookup %q map: %s", mapName, err)
+	}
+	return pgKernelTelemetry, nil
+}
+
+// compareMessagesCount returns true if the expected bucket is non-empty
+// each buckets may have 1 or 2 hits
```

Comment:
> each buckets may have 1 or 2 hits

what do you mean?
that might be relevant for a single test, but not true for the general case